src/Playroom/Snippets/Snippets.tsx

```diff
@@ -25,26 +25,48 @@ function getSnippetId(snippet: Snippet, index: number) {
   return `${snippet.group}_${snippet.name}_${index}`;
 }
 
-const filterSnippetsForTerm = (snippets: Props['snippets'], term: string) =>
-  term
-    ? fuzzy
-        .filter(term, snippets, {
-          extract: (snippet) => `${snippet.group} ${snippet.name}`,
-        })
-        .map(({ original, score }) => ({ ...original, score }))
-    : snippets;
+const options = {
+  threshold: 0.3,
+  keys: [
+    {
+      name: 'group',
+      weight: 2,
+    },
+    {
+      name: 'name',
+      weight: 1,
+    },
+  ],
+};
 
 export default ({ isOpen, snippets, onHighlight, onClose }: Props) => {
   const [searchTerm, setSearchTerm] = useState<string>('');
   const [highlightedIndex, setHighlightedIndex] =
     useState<HighlightIndex>(null);
+
   const listEl = useRef<HTMLUListElement | null>(null);
   const highlightedEl = useRef<HTMLLIElement | null>(null);
+
+  const fuse = useMemo(() => new Fuse(snippets, options), [snippets]);
+  const fuseResults = useMemo(
+    () => (searchTerm ? fuse.search(searchTerm) : []),
+    [fuse, searchTerm]
+  );
+
+  const filteredSnippets = useMemo(
+    () =>
+      fuseResults.length > 0
+        ? fuseResults.map((result) => result.item)
+        : snippets,
+    [fuseResults, snippets]
+  );
```

Comment:
Since the second memo is dependent on the first one, could these be joined together?